readthedocs/api/v2/views/integrations.py

```diff
@@ -359,21 +359,23 @@ def get_digest(secret, msg):
     def handle_webhook(self):
         # Get event and trigger other webhook events
         action = self.data.get('action', None)
+        deleted = self.data.get('deleted', False)
         event = self.request.META.get(GITHUB_EVENT_HEADER, GITHUB_PUSH)
         webhook_github.send(
             Project,
             project=self.project,
             data=self.data,
             event=event,
         )
-        # Handle push events and trigger builds
-        if event == GITHUB_PUSH:
+        # Don't build a branch if it's a push that was actually a delete
+        # https://developer.github.com/v3/activity/events/types/#pushevent
+        if event == GITHUB_PUSH and not deleted:
```

Comment:
If `created=True` this wouldn't trigger a new build, because the version doesn't exist yet in read the docs.

https://github.com/readthedocs/readthedocs.org/blob/2e67b7d51ff10eb8e422c6945949257685c35d40/readthedocs/core/views/hooks.py#L50-L52

We should check that created is `False` here too.

Reply:
That's an existing bug though, right?

Reply:
Yeah, a push event + deleted or push event + created should trigger a version sync, not a build (because it will fail when deteled=True and it wouldn't to anything if created=True)

Reply:
Ah, I understand, so we need to check for `created` on the push as well 👍